.github/workflows/schemas.yml

```diff
@@ -1,52 +1,61 @@
 name: Schemas
+run-name: Update schema for ref ${{ github.event.client_payload.ref || inputs.ref || 'main' }}
 
 on:
-  # Activate this workflow manually
-  workflow_dispatch:
+  workflow_dispatch: # Activate this workflow manually
+    inputs:
+      ref:
+        description: Tag or branch name of Haystack version
+        required: true
+        type: string
+        default: main
 
   # Activate this workflow with an API call
   # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch
   repository_dispatch:
     types: [generate-pipeline-schemas]
 
+env:
+  HAYSTACK_REF: ${{ github.event.client_payload.ref || inputs.ref || 'main' }}
+  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
 jobs:
 
   run:
+    name: Update schema for ref ${{ github.event.client_payload.ref || inputs.ref || 'main' }}
     runs-on: ubuntu-latest
     steps:
 
-      - name: Checkout
-        uses: actions/checkout@v3
-
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.7
+      - name: Validate release
+        id: validate
+        continue-on-error: true
```

Comment:
Curious why we don't just fail here?

Reply:
The idea is to silently discard docker releases for test purposes like https://github.com/deepset-ai/haystack/actions/runs/3756768793 to not publish schema files for them, while avoiding a red CI.

We can also move this filter to the haystack repo, closer to the origin of the workflow dispatch if that would make more sense.

Reply:
I'm fine with both, can you add a comment explaining it for our future self? (Basically just pasting the comment above somewhere in the code)